### ISSUE TITLE
Feature Request: Aoto-create the group token

### ISSUE THREAD

**Issue by coffiarts** [coffiarts/FoundryVTT-crunch-my-party#4]
I am posting this myself as a possible feature I might be working on soon, as there has been various feedback about this. 
Many people would like to see this mod being able to automatically create a group token from its member tokens.
There's been discussions about this on youtube, e.g. [here](https://www.youtube.com/watch?v=F8f29g0SLqA&lc=UgxUHPx_VVpmCqlE11Z4AaABAg.9ugVrxWMHYx9ukneSL3T4A).

Hitherto, having to create a dedicated token for the group is not a very elegant approach. 
Usually my recommended way is to use [Tokenizer](https://github.com/mrprimate/tokenizer/), for quickly assembling your party members' avatar images into a nice combined token, [as demonstrated here](https://raw.githubusercontent.com/MrPrimate/tokenizer/master/docs/mask.gif). Yet I uncerstand that not everyone is happy having to do this, especially when the list of members changes frequently.

I assume by now that this _should_ be doable in a similar same way as I've managed to generate overlay icons in my [Lock The Sheets!](https://github.com/coffiarts/FoundryVTT-lock-the-sheets) module. But it will require some research and experimenting!

**Comment by eddiecooro** [comment]
Hey there @coffiarts 👋 Hope you're doing well!

First of all, thanks for this amazing package, and also for those fun introduction videos—I really enjoyed your marketing skills! XD

I wanted to ask about the status of this repo. I noticed that your latest commit was around 4 months ago. I hope that’s due to the stability of the package rather than a lack of interest in continuing development. :(

I’m particularly interested in this specific feature and am willing to open a PR for it if you’re open to it. I’ve already got a POC that took ages to get working properly, and I’d love to move forward with it, but I don’t want to spend more time without knowing if it’s likely to be merged. 

Thanks in advance!

**Comment by coffiarts** [comment]
Hi @eddiecooro!
I was absolutely amazed reading your lines! Be assured that I _am_ still continuing this mod. BTW note that my most recent commit has been on 11th July (https://github.com/coffiarts/FoundryVTT-crunch-my-party/releases/tag/latest), i.e. not "4 months ago" ;-).
Apart from that, you're absolutely right concerning this specific change request: I have been silent about it since I created it. This is simply because I just haven't found the right approach, neither the time so far to fully delve into it. So I've postponed it over and over again.
All the more I am **extremely enthusiastic** to learn that you've picked up the pieces here and seem to have found a solution!
I'll happily look into your PR and support it the best I can.
Looking forward to your code!

**Comment by eddiecooro** [comment]
You're right you're right, I misread the latest commit on this repo—it's actually from 1 month ago, not 4.

Regarding the MR, I already have a working proof of concept, and I’ll try to integrate it into the codebase over the next week or so. Here’s how it looks so far:

https://github.com/user-attachments/assets/e2e5de6f-76ea-4783-bd04-9db35a4eeb37

<sub>(Had to reduce the quality because of github limitations)</sub>

I have to say, it was much more challenging to achieve than I expected! 😅

Only the integration part remains. Here’s my plan for what to do—thought I’d run it by you first:

1. When creating a party group, add an option next to "Enter the name of token" to auto-generate a token.
2. Generate a token, assign the image, and store the ID instead of the token name.
3. Proceed with the rest of the "Crunch my party" logic, using the generated token as the baseline.

This isn’t complete, but it should suffice for now. Once this is solid and merged, I plan to work on a few more things.

First, I’d like to introduce a new token image merge mode called "Stack," which would be useful for top-down tokens and similar tokens to the ones used in your demos. This should be easier than the combine mode I’ve already implemented.

Second, I’d like to allow these kinds of tokens to be created on the fly when toggling the party, potentially eliminating the need for the "group" macro altogether. If a token associated with the selected tokens is found, the toggle happens; otherwise, the create modal will pop up.

Third, though it needs some research, I’d love to be able to store the token data somehow. Creating an actor automatically could work, but system differences would make it difficult to maintain. I’m thinking about storing the serialized version of non-linked tokens and the actor ID of the linked ones.

This would also remove the need to manually add the group token to the scene or disable/enable it, as it would use the stored ID to create a new token each time. It would also enable us to implement more features later on.

Beyond these, there are a few other improvements I’d love to see at some point, though I’m not sure I’ll have the time to work on them. Here they are, in no particular order:

- Introduce options to tweak the configs of the image generation process, such as the order of tokens around the circle, the amount of blend between them (how much tokens overlap), the scale multiplier, and maybe an offset for each token. The image generation code is already designed with this in mind, but we’d need to implement the UI for it.
- This one’s a bit far-fetched, but in an ideal world, we’d implement a new Actor subtype called "Group." It could be created manually, allowing configurations to be added to each group, stored in the actor tree, and automatically generating the group image on the fly when tokens are added or removed. It wouldn’t need any macros anymore, would have a better UI, and would enable features like "marching order" to be implemented.

Sorry for the long message—just wanted to share all the ideas that have been bouncing around in my head! I’m really excited about the potential for these improvements and would love to hear your thoughts on them. :)

**Comment by coffiarts** [comment]
OMG... this looks veeeeeery promising. I'll need some time to go deeper into all your ideas, but I guess I have some first ideas for some of them (concerning the integration). Sounds like a bigger roadmap, but it will be worth going that way to evolve my mod to the next level.
